packages/jest-runtime/src/index.ts

```diff
@@ -416,12 +418,15 @@ export default class Runtime {
     query = '',
   ): Promise<VMModule> {
     const cacheKey = modulePath + query;
+    const registry = this._isolatedESMModuleRegistry
+      ? this._isolatedESMModuleRegistry
+      : this._esmoduleRegistry;
```

Comment:
Hm.. Isn’t it enough to use the same `_isolatedMockRegistry`? Did you try that?

It was my idea about missing `_isolatedESMModuleRegistry`, but it can I was wrong. Here we can see that `_isolatedMockRegistry` was simply not used while loading ES modules. Sorry about slow thinking. Could you try to `_isolatedMockRegistry` everywhere?

Reply:
Of course, it is possible. I originally thought it was intentionally differentiated because I saw that there was already an existing `_isolatedModuleRegistry`, moreover, I think it seems more appropriate.